### ISSUE TITLE
How do you see the generated module names?

### ISSUE THREAD

**Issue by mpdairy** [google/proto-lens#328]
I have some complicated protos I'm trying to work with and would find it very helpful to see a list of the modules that are created from the proto files (or, at least, that would be created if I knew their names and could put them in `exposed-modules`). Is there some way to get a list of them?

**Comment by judah** [comment]
Unfortunately there isn't an automated way to get the names.  It's probably something we could improve, for example as another flag to the proto-lens-protoc binary, or in the stderr of the setup script's output.  We could also probably improve this in the documentation.

The logic is intentionally pretty simple:
https://github.com/google/proto-lens/blob/0cb384a09810b4d512a432106e7a9602d2e2e7a6/proto-lens-protoc/src/Data/ProtoLens/Compiler/Plugin.hs#L94

In short:
- Each path component of the filepath gets turned from `snake_case` into `CamelCase` (dropping the `.proto` extension).
- The primary module is those components together with `.`, plus a `Proto.` prefix.
- There's also another (optional) module with `_Fields` appended.

So for example, `foo/bar_baz.proto` turns into `Proto.Foo.BarBaz` and `Proto.Foo.BarBaz_Fields`.

**Comment by mpdairy** [comment]
Ok, thanks. I was thinking the `Proto` prefix came from the `proto` folder, which led to much confusion.

Is it true that you can only run `defaultMainGeneratingProtos "proto"` once in `Setup.hs`? I was writing a program that translates messages between two sets of proto files, and was forced to put them all in the same `proto` folder, which works for now, but it would be nicer if I could keep them separate.

**Comment by judah** [comment]
Yes, currently it only allows a single "root" directory per project.  Two possible workarounds are:

- Create different Cabal libraries for the the different folders
- Symlink the different folders into one common location (which would add an additional prefix component to the module names).
